src/compute/src/render/join/linear_join.rs

```diff
@@ -312,3 +370,435 @@ where
         (oks, None)
     }
 }
+
+/// Joins two arranged collections with the same key type.
+///
+/// Each matching pair of records `(key, val1)` and `(key, val2)` are subjected to the `result` function,
+/// which produces something implementing `IntoIterator`, where the output collection will have an entry for
+/// every value returned by the iterator.
+fn mz_join_core<G, Tr1, Tr2, L, I>(
```

Comment:
Maybe it's worth moving this to a separate module or file, so we can have a higher-level comment explaining why we're adding our own version of `join_core`, and to keep it more separate from uses.

Reply:
Good idea!